src/Workspaces/CSharp/Portable/Classification/ClassificationHelpers.cs

```diff
@@ -118,6 +119,48 @@ private static string GetClassificationForIdentifier(SyntaxToken token)
             {
                 return ClassificationTypeNames.TypeParameterName;
             }
+            else if (token.Parent is MethodDeclarationSyntax methodDeclaration && methodDeclaration.Identifier == token)
+            {
+                // TODO: Add proper extension method check - I'm sure there is some syntax based check somewhere burried in roslyn but I couldn't find it.
+                bool isExtensionMethod =
+                    (methodDeclaration.Parent as TypeDeclarationSyntax)?.Modifiers.Any(m => m.IsKind(SyntaxKind.StaticKeyword)) == true &&
+                    methodDeclaration.Modifiers.Any(m => m.IsKind(SyntaxKind.StaticKeyword)) &&
+                    methodDeclaration.ParameterList.Parameters.FirstOrDefault()?.Modifiers.Any(m => m.IsKind(SyntaxKind.ThisKeyword)) == true;
+
+                return isExtensionMethod ? ClassificationTypeNames.ExtensionMethodName : ClassificationTypeNames.MethodName;
+            }
+            else if (token.Parent is PropertyDeclarationSyntax propertyDeclaration && propertyDeclaration.Identifier == token)
+            {
+                return ClassificationTypeNames.PropertyName;
+            }
+            else if (token.Parent is EnumMemberDeclarationSyntax enumMemberDeclaration && enumMemberDeclaration.Identifier == token)
+            {
+                return ClassificationTypeNames.EnumFieldName;
+            }
+            else if (token.Parent is VariableDeclaratorSyntax variableDeclarator && variableDeclarator.Identifier == token)
+            {
+                var varDecl = variableDeclarator.Parent as VariableDeclarationSyntax;
+                switch (varDecl.Parent)
+                {
+                    case FieldDeclarationSyntax fieldDeclaration:
+                        return fieldDeclaration.Modifiers.Any(m => m.IsKind(SyntaxKind.ConstKeyword)) ? ClassificationTypeNames.ConstantName : ClassificationTypeNames.FieldName;
```

Comment:
don't use lambda overload.  don't use linq.  wrap long line.  align ? and : of ternary on next line.